integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+     *
+     * @see #supported()
+     *
+     * @see #create(String, Config)
+     *
+     * @deprecated For use by the Helidon Config subsystem only.
+     */
+    @Deprecated // For use by the Helidon Config subsystem only.
+    @Override // ConfigSourceProvider
+    public boolean supports(String type) {
+        return type != null && this.supported().contains(type);
+    }
+
+    /**
+     * Returns a (determinate) weight ({@value #WEIGHT}) for this {@link OciSecretsConfigSourceProvider} when invoked.
+     *
+     * @return a determinate weight ({@value #WEIGHT}) when invoked
+     *
+     * @deprecated For use by the Helidon service loading subsystem only.
+     */
+    @Deprecated // For use by the Helidon service loading subsystem only.
+    @Override // Weighted
+    public double weight() {
+        return WEIGHT;
+    }
+
+
+    /*
+     * Inner and nested classes.
+     */
+
+
+    private static final class SecretBundleConfigSource extends AbstractConfigSource implements NodeConfigSource, PollableSource<Instant> {
+
+
+        /*
+         * Static fields.
+         */
+
+
+        private static final Optional<NodeContent> ABSENT_NODE_CONTENT =
+            Optional.of(NodeContent.builder().node(ObjectNode.empty()).build());
+
+        private static final String COMPARTMENT_OCID_PROPERTY_NAME = "compartment-ocid";
+
+        private static final Logger LOGGER = System.getLogger(SecretBundleConfigSource.class.getName());
+
+        private static final String VAULT_OCID_PROPERTY_NAME = "vault-ocid";
+
+
+        /*
+         * Instance fields.
+         */
+
+
+        private final Supplier<? extends Optional<NodeContent>> loader;
+
+        private volatile Instant mostDistantExpirationInstant;
+
+
+        /*
+         * Constructors.
+         */
+
+
+        @SuppressWarnings("try")
+        private SecretBundleConfigSource(Builder b) {
+            super(b);
+            Supplier<? extends Secrets> secretsSupplier = Objects.requireNonNull(b.secretsSupplier, "b.secretsSupplier");
+            Supplier<? extends Vaults> vaultsSupplier = Objects.requireNonNull(b.vaultsSupplier, "b.vaultsSupplier");
+            String compartmentOcid = b.compartmentOcid;
+            String vaultOcid = b.vaultOcid;
+            if (compartmentOcid == null || vaultOcid == null) {
+                // (It is not immediately clear why the OCI Java SDK requires a Compartment OCID, since a Vault OCID is
+                // sufficient to uniquely identify any Vault.)
+                this.loader = this::absentNodeContent;
+            } else {
+                ListSecretsRequest listSecretsRequest = ListSecretsRequest.builder()
+                    .compartmentId(compartmentOcid)
+                    .vaultId(vaultOcid)
+                    .build();
+                this.loader = () -> {
+                    Collection<? extends SecretSummary> secretSummaries = secretSummaries(vaultsSupplier, listSecretsRequest);
```

Comment:
Can you please make this a private method - seems a bit long to inline like this.

Reply:
It closes over various variables. Do you want me to turn them into (less efficient, unneeded in some cases) instance variables?

Reply:
Oh sorry, now I see what you are saying. Will change.

Reply:
Done